Mage.Sets/src/mage/cards/t/TrenzaloreClocktower.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.t;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.costs.common.ExileSourceCost;
+import mage.abilities.costs.common.RemoveCountersSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.decorator.ConditionalActivatedAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.mana.BlueManaAbility;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.counters.CounterType;
+import mage.filter.common.FilterControlledPermanent;
+import mage.game.Game;
+import mage.players.Player;
+
+/**
+ * @author Cguy7777
+ */
+public final class TrenzaloreClocktower extends CardImpl {
+
+    private static final FilterControlledPermanent filter
+            = new FilterControlledPermanent(SubType.TIME_LORD, "you control a Time Lord");
+
+    public TrenzaloreClocktower(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.LAND}, "");
+
+        this.supertype.add(SuperType.LEGENDARY);
+
+        // {T}: Add {U}. Put a time counter on Trenzalore Clocktower.
+        Ability manaAbility = new BlueManaAbility();
+        manaAbility.addEffect(new AddCountersSourceEffect(CounterType.TIME.createInstance()));
+        this.addAbility(manaAbility);
+
+        // {1}{U}, {T}, Remove twelve time counters from Trenzalore Clocktower and exile it:
+        // Shuffle your graveyard and hand into your library, then draw seven cards.
+        // Activate only if you control a Time Lord.
+        Ability wheelAbility = new ConditionalActivatedAbility(
+                new TrenzaloreClocktowerEffect(),
```

Comment:
best to use a common effect for the draw 7 rather than lumping it in to the prior custom effect